Address ID review

AustinSeto opened the pull request:
Closes #21

Commit:
Addresses ID review of What You'll Learn section

Commit:
Update "Enabling GraphQL" section

Commit:
Address ID review of "Running Application" section

Commit:
Address ID review of testing section

Commit:
Address ID review of resolvers section

Commit:
Address ID review of object types section

Rwalls1 commented on README.adoc:
```diff
@@ -32,27 +32,29 @@ Explore how to use GraphQL to access and manipulate data using MicroProfile Grap
-This is beneficial when dealing with large resources or ones that contain properties that are expensive to calculate. 
+GraphQL is an open source data query language. 
+Unlike REST APIs, each `POST` request that is sent to a GraphQL service goes to a single HTTP endpoint.
+CRUD operations and their details are differentiated by the contents of the request.
```
You should spell out CRUD on first reference to maintain clarity, so change "CRUD operations" to "Create, read, update, and delete (CRUD) operations".

Rwalls1 commented on README.adoc:
```diff
@@ -32,27 +32,29 @@ Explore how to use GraphQL to access and manipulate data using MicroProfile Grap
+GraphQL is an open source data query language. 
+Unlike REST APIs, each `POST` request that is sent to a GraphQL service goes to a single HTTP endpoint.
+CRUD operations and their details are differentiated by the contents of the request.
+If the operation returns data, the user will also specify what properties of the data that they want returned. 
```
Change "will also specify" to "specifies" to avoid future tense.

Rwalls1 commented on README.adoc:
```diff
@@ -32,27 +32,29 @@ Explore how to use GraphQL to access and manipulate data using MicroProfile Grap
+Unlike REST APIs, each `POST` request that is sent to a GraphQL service goes to a single HTTP endpoint.
+CRUD operations and their details are differentiated by the contents of the request.
+If the operation returns data, the user will also specify what properties of the data that they want returned. 
+For read operations, a JSON object is returned that contains only the data and properties specified.
```
Typo: Change "properties specified" to "properties that are specified"

Rwalls1 commented on README.adoc:
```diff
@@ -32,27 +32,29 @@ Explore how to use GraphQL to access and manipulate data using MicroProfile Grap
+CRUD operations and their details are differentiated by the contents of the request.
+If the operation returns data, the user will also specify what properties of the data that they want returned. 
+For read operations, a JSON object is returned that contains only the data and properties specified.
+For other operations, a JSON object may be returned containing information such as a success message. 
```
Change "may be returned" to "might be returned" to maintain clarity.

Rwalls1 commented on README.adoc:
```diff
@@ -116,21 +117,21 @@ You'll apply various annotations to the properties and declaration of a Java cla
 The [hotspot=class file=0]`JavaInfo` class is annotated with an `@Type` annotation. 
 The [hotspot=type file=0]`@Type("java")` annotation maps this class to define the `java` object type in GraphQL.
-The `java` object type will give information on the system's Java installation. 
+The `java` object type gives information on the system's Java installation. 
```
Change "the system's Java installation" to "the Java installation of the system" to avoid giving possessive traits to entities that aren't referring to users.

Rwalls1 commented on README.adoc:
```diff
@@ -116,21 +117,21 @@ You'll apply various annotations to the properties and declaration of a Java cla
 The [hotspot=getVendor file=0]`getVendor()` and [hotspot=getVersion file=0]`getVersion()` getter functions 
-will automatically be mapped to retrieve their respective properties in GraphQL. 
+are automatically mapped to retrieve their respective properties in GraphQL. 
 If needed, setter functions are also supported and will be automatically mapped. 
```
Change "will be automatically mapped" to "automatically mapped" to avoid future tense.

Rwalls1 commented on README.adoc:
```diff
@@ -190,24 +191,24 @@ The `mutation` operation type can create, delete, or modify data.
 
 The resolvers are defined in the [hotspot file=0]`SystemResource.java` file.
-The [hotspot=graphqlapi file=0]`@GraphQLApi` annotation makes it so that GraphQL will use the methods defined in this class as resolvers. 
+The [hotspot=graphqlapi file=0]`@GraphQLApi` annotation enables GraphQL to use the methods defined in this class as resolvers. 
```
Typo: Change "methods defined" to "methods that are defined"

Rwalls1 commented on README.adoc:
```diff
@@ -263,15 +264,15 @@ include::{common-includes}/devmode-build.adoc[]
 GraphiQL has already been set up and included for you.
 Access GraphiQL at the http://localhost:9080/graphiql.html[^] URL.
-Queries made through GraphiQL will be the same as queries made through HTTP requests. 
+Queries made through GraphiQL are the same as queries made through HTTP requests. 
```
Typo: Change "Queries made through" to "Queries that are made through"

Rwalls1 commented on README.adoc:
```diff
@@ -285,7 +286,7 @@ query {
 ----
 
-The output will be similar to the following:
+The output will be similar to the following example:
```
Change "will be similar" to "is similar"

Rwalls1 commented on README.adoc:
```diff
@@ -360,12 +361,12 @@ The [hotspot file=0]`SystemIT` class contains the test methods.
-It will try to set the note to be the current system time. 
-Then, it makes a query using the `system` resolver to ensure the note was saved correctly. 
+It tries to set the note to be the current system time. 
+Then, it makes a query by using the `system` resolver to ensure the note was saved correctly. 
```
Typo: Change "ensure the note" to "ensure that the note"

Rwalls1 reviewed (commented):
The updates look good, I just had a few more comments.

Commit:
Address PR comments

Commit:
Rewrite paragraphs that describe sections in future tense

AustinSeto commented:
Updated guide with suggested changes and rewrote two paragraphs in style of https://github.com/OpenLiberty/draft-guide-graphql-intro/issues/21#issuecomment-782374557, lgdev site is updated with latest changes

Rwalls1 commented on README.adoc:
```diff
@@ -105,8 +106,8 @@ Navigate to the `start` directory to begin.
-You'll create three object types to define the structure of the data returned by GraphQL.
-You'll apply various annotations to the properties and declaration of a Java class to define an object type.
+Object types define the structure of the data returned by GraphQL. 
+Annotations applied to the declaration and properties of Java classes define object types.
```
Typo: Change "Annotations applied" to "Annotations that are applied"

Rwalls1 commented on README.adoc:
```diff
@@ -179,7 +180,8 @@ include::finish/src/main/java/io/openliberty/guides/graphql/models/SystemInfo.ja
 Resolvers are functions that provide instructions for GraphQL operations.
-We'll implement resolvers for the `query` and `mutation` operation types.
+Each operation requires a corresponding resolver.
+This guide contains an example of a `query` operation, and a `mutation` operation. 
```
Remove this entire sentence to be more concise. Within the context of the guide, the sentence doesn't add any value because the guide later specifically mentions those operation types so the sentence is already implied.

Rwalls1 commented on README.adoc:
```diff
@@ -263,15 +265,15 @@ include::{common-includes}/devmode-build.adoc[]
 GraphiQL has already been set up and included for you.
 Access GraphiQL at the http://localhost:9080/graphiql.html[^] URL.
-Queries made through GraphiQL will be the same as queries made through HTTP requests. 
+Queries that are made through GraphiQL are the same as queries made through HTTP requests. 
```
Typo: Change "queries made through HTTP requests" to "queries that are made through HTTP requests"

Rwalls1 commented on README.adoc:
```diff
@@ -263,15 +265,15 @@ include::{common-includes}/devmode-build.adoc[]
-If any data types are mentioned, you'll also be able to see descriptions of them and the fields contained within them if applicable. 
-The schema will also be visible at the http://localhost:9080/graphql/schema.graphql[^] URL. 
+The documentation describes the GraphQL schema.
+The schema gives details on the various operations available. 
```
Typo: Change "operations available" to "operations that are available"

Rwalls1 commented on README.adoc:
```diff
@@ -263,15 +265,15 @@ include::{common-includes}/devmode-build.adoc[]
-Queries made through GraphiQL will be the same as queries made through HTTP requests. 
+Queries that are made through GraphiQL are the same as queries made through HTTP requests. 
 
 In GraphiQL, you can view the application documentation by clicking `Docs`. 
```
Change "`Docs`" to something like "the `Docs` button in the navigation menu" to clarify what action is taking place

Rwalls1 reviewed (commented):
@AustinSeto Thanks, I just have a few more comments.

Commit:
Address most recent bit of ID review

AustinSeto commented:
Updated with suggested changes, changes pushed to lgdev site

Rwalls1 reviewed (commented):
@AustinSeto The latest changes look good, I just have a couple more comments:
## What you’ll learn
Typo: Change "properties specified" to "properties that are specified"
## Creating GraphQL object types
- Change "object types will be used" to "object types are used" to avoid future tense.
- Change "object type's class" to "class of the object type" to avoid using the possessive phrase to reference the object type
## Running the tests
- To avoid future tense, remove the phrase "You will see the following output:" and rephrase the first paragraph to say, "Because you started Open Liberty in dev mode, press the enter/return key to run the tests that show the following output:”

Commit:
Address ID suggestions

Commit:
Merge branch 'draft' of github.com:OpenLiberty/draft-guide-graphql-intro into address-id-review

# Conflicts:
#	README.adoc

AustinSeto commented:
Updated with the latest suggestions. We also updated the testing section in #23, which included some minor additions to the testing section. All the changes are shown on our lgdev site.

Rwalls1 reviewed (approved):
The changes to the file look good, however I don't see the changes on the lgdev site: http://lgdev1.fyre.ibm.com:4020/guides/graphql-intro.html#what-youll-learn
